src/fr/animesama/src/eu/kanade/tachiyomi/extension/fr/animesama/AnimeSama.kt

```diff
@@ -135,6 +140,54 @@ class AnimeSama : ParsedHttpSource() {
         return GET(url, headers)
     }
 
+    private fun detailsParse(response: Response, id: String): SManga? {
+        val document = response.asJsoup()
+
+        val scriptContent = document.select("script:containsData(panneauScan(\"nom\", \"url\"))").toString()
+        val splitedContent = scriptContent.split(";").toMutableList()
+        // Remove exemple
+        splitedContent.removeAt(0)
+
+        val parsedChapterList: MutableList<SChapter> = mutableListOf()
+        val numberOfScans = splitedContent.count { line ->
+            val pattern = """panneauScan\("(.+?)", "(.+?)"\)""".toRegex()
+            val matchResult = pattern.find(line)
+            matchResult != null && !matchResult.groupValues[2].contains("va")
+        }
+
+        if (numberOfScans == 0) return null
+
+        val manga: SManga = SManga.create()
+        manga.description = document.select("#sousBlocMiddle > div h2:contains(Synopsis)+p").text()
+        manga.genre = document.select("#sousBlocMiddle > div h2:contains(Genres)+a").text()
+        manga.title = document.select("#titreOeuvre").text()
+        manga.thumbnail_url = document.select("#coverOeuvre").attr("src")
+        manga.setUrlWithoutDomain(document.baseUri())
+        return manga
+    }
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        return if (query.startsWith("ID:")) {
+            val id = query.substringAfterLast("ID:")
+
+            val mangaUrl = baseUrl.toHttpUrl()
+                .newBuilder()
+                .query(null)
+                .addPathSegment("catalogue")
+                .addPathSegment(id)
+                .build()
```

Comment:
Not sure why you're setting `query` to `null` when `baseUrl` doesn't have one already.
```suggestion
            val mangaUrl = "$baseUrl/catalogue".toHttpUrl()
                .newBuilder()
                .addPathSegment(id)
                .build()
```